python/ray/dag/dag_node.py

```diff
@@ -167,6 +185,7 @@ def experimental_compile(
         if _max_buffered_results is None:
             _max_buffered_results = ctx.max_buffered_results
 
+        self.is_output_node = True
```

Comment:
can we remove this with preprocess update?

Reply:
No, in `preprocess`, it currently assumes that a DAG node with no downstream nodes is the output node. However, in this PR, both the output node and leaf nodes have no downstream nodes.

https://github.com/ray-project/ray/blob/ea8b8e186c3a6384429e0990faf85e485dbbdada/python/ray/dag/compiled_dag_node.py#L850